Mathlib/NumberTheory/Transcendental/Lindemann/Init/AnalyticalPart.lean

```diff
@@ -0,0 +1,206 @@
+/-
+Copyright (c) 2022 Yuyang Zhao. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yuyang Zhao
+-/
+import Mathlib.Algebra.Polynomial.SumIteratedDerivative
+import Mathlib.Analysis.Calculus.Deriv.Polynomial
+import Mathlib.Analysis.SpecialFunctions.ExpDeriv
+import Mathlib.MeasureTheory.Integral.FundThmCalculus
+import Mathlib.RingTheory.Int.Basic
+import Mathlib.Topology.Algebra.Polynomial
+
+/-!
+# The Lindemann-Weierstrass theorem
+-/
+
+namespace LindemannWeierstrass
+
+noncomputable section
+
+open scoped Nat
+
+open Complex Polynomial
+
+attribute [local fun_prop] Polynomial.differentiable Complex.continuous_abs
+
+theorem hasDerivAt_cexp_mul_sumIDeriv (p : ℂ[X]) (s : ℂ) (x : ℝ) :
+    HasDerivAt (fun x : ℝ ↦ -(cexp (-(x • s)) * p.sumIDeriv.eval (x • s)))
+      (s * (cexp (-(x • s)) * p.eval (x • s))) x := by
+  have h₀ := (hasDerivAt_id' x).smul_const s
+  have h₁ := h₀.neg.cexp
+  have h₂ := ((sumIDeriv p).hasDerivAt (x • s)).comp x h₀
+  convert (h₁.mul h₂).neg using 1
+  nth_rw 1 [sumIDeriv_eq_self_add p]
+  simp only [one_smul, eval_add, Function.comp_apply]
+  ring
+
+theorem integral_exp_mul_eval (p : ℂ[X]) (s : ℂ) :
+    s * ∫ x in (0)..1, exp (-(x • s)) * p.eval (x • s) =
+      -(exp (-s) * p.sumIDeriv.eval s) + p.sumIDeriv.eval 0 := by
+  rw [← intervalIntegral.integral_const_mul,
+    intervalIntegral.integral_eq_sub_of_hasDerivAt
+      (fun x hx => hasDerivAt_cexp_mul_sumIDeriv p s x)
+      (ContinuousOn.intervalIntegrable (by fun_prop))]
+  simp
+
+private def P (p : ℂ[X]) (s : ℂ) :=
```

Comment:
Even though (or maybe exactly because) this is a private definition, I think it is good to add a docstring explaining the raison d'être of this `P`.